cypress/support/api_commands.js

```diff
@@ -258,31 +276,23 @@ Cypress.Commands.add('createNewUser', (user = {}, teams = null) => {
 
         const userId = userResponse.body.id;
 
-        let teamsToAdd;
-
-        if (teams) {
-            teamsToAdd = teams;
+        if (teamIds && teamIds.length > 0) {
+            teamIds.forEach((teamId) => {
+                cy.apiAddUserToTeam(teamId, userId);
+            });
         } else {
             // Get teams, select the first three, and add new user to that team
             cy.request('GET', '/api/v4/teams').then((teamsResponse) => {
                 // Verify we have at least 2 teams in the response to add the user to
                 expect(teamsResponse).to.have.property('body').to.have.length.greaterThan(1);
 
                 // Pull out only the first 2 teams
-                teamsToAdd = teamsResponse.body.slice(0, 2).map((t) => t.id);
+                teamsResponse.body.slice(0, 2).map((t) => t.id).forEach((teamId) => {
+                    cy.apiAddUserToTeam(teamId, userId);
+                });
             });
         }
 
-        // Add user to several teams by default
-        teamsToAdd.forEach((team) => {
```

Comment:
This is throwing error in some test.  Seems that this line is being called asap and not awaiting from the `teamsResponse` above.